src/main/java/edu/uwaterloo/ece658/service/AccountService.java

```diff
@@ -94,12 +83,37 @@ public void updateAdditionalInformation(String userName,
 
     public void updateAdditionalInformation(String userName, boolean isAdmin,
             String newFName, String newLName, String newGender, int newAge) {
-        assert exist(userName, isAdmin);
+        assert userFacade.existUserWithRole(userName, isAdmin);
```

Comment:
is using assert safe? What happens if it fails ? On that note, should we be throwing and catching exceptions during these circumstances?

Reply:
Using assert is not something I like. We need to have different exceptions for these cases. But that's steps later. So assert's here only to say that you need to improve it later on.